### PR TITLE
tord binary search tree fp

### PR DESCRIPTION
**Pull request by Kazads**

### CODE REVIEW

**Review by emilfolino — COMMENTED** [review]
Hej kan jag får göra detta som ett test?

File: FPBinarySearchTreeTord/es6-test/bst_test.js [hunk 1]
```diff
@@ -0,0 +1,81 @@
+import {createTree} from '../dist/js/BinarySearchTree';
+import {insert} from '../dist/js/BinarySearchTree';
+import {findNode} from '../dist/js/BinarySearchTree';
+import {inOrderTraversal} from '../dist/js/BinarySearchTree';
+import { expect } from 'chai';
+
+describe('BinarySearchTree', function() {
```
Comment (KimSvenSand) [inline FPBinarySearchTreeTord/es6-test/bst_test.js:7]: With correct test data

File: FPBinarySearchTreeTord/src/index.js [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+  var tree = [];
+  let inputArray = [];
+  var treelength = 0;
+  while(treelength < 10){
+    var tree = insert(Math.floor((Math.random()*100)+1),tree);
+    iterations++
+  }
+  console.log("Unsorted: "+tree);
+  console.log("Sorted: "+inOrderTraversal(tree,0,[]));
+
```
Comment (KimSvenSand) [inline FPBinarySearchTreeTord/src/index.js:24]: This again :octocat:

File: FPBinarySearchTreeTord/src/js/BinarySearchTree.js [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+  }else if(input == tree[place]){
+    returnValue = tree[place];
+  }else if(input < tree[place]){
+    returnValue = findNode(input,tree,(place*2+1));
+  }else{
+    returnValue = findNode(input,tree,(place*2+1));
+  }
+  return returnValue;
+}
+
```
Comment (KimSvenSand) [inline FPBinarySearchTreeTord/src/js/BinarySearchTree.js:51]: You're mutating returnArray. Don't! It's so sad :sake:

File: FPBinarySearchTreeTord/src/js/BinarySearchTree.js [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+    }
+  }
+  return returnArray;
+}
+
+export function insert(input,tree){
+  var returnArray = findPlaceForInput(tree,0,input);
+  return returnArray;
+}
+
```
Comment (KimSvenSand) [inline FPBinarySearchTreeTord/src/js/BinarySearchTree.js:38]: Mutations :shipit:

File: FPBinarySearchTreeTord/src/js/BinarySearchTree.js [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+  }else{
+    if(input < returnArray[place]){
+      returnArray = findPlaceForInput(returnArray,(place * 2 +1),input);
+    }else if(input > returnArray[place]){
+      returnArray = findPlaceForInput(returnArray,(place * 2 + 2),input);
+    }
+  }
+  return returnArray;
+}
+
```
Comment (KimSvenSand) [inline FPBinarySearchTreeTord/src/js/BinarySearchTree.js:33]: This just might be a bit unessecary :octocat:

Comment (KimSvenSand) [reply FPBinarySearchTreeTord/src/js/BinarySearchTree.js:33]: KILLED IT!

File: FPBinarySearchTreeTord/src/js/BinarySearchTree.js [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+export function inputArrayInTree(inputArray,placeInInputArray,returnArray){
+    if(inputArray.length-1 >= placeInInputArray){
+      returnArray = findPlaceForInput(returnArray,0,inputArray[placeInInputArray]);
+      returnArray = inputArrayInTree(inputArray,(placeInInputArray+1),returnArray);
+    }
+  return returnArray;
+}
+
```
Comment (KimSvenSand) [inline FPBinarySearchTreeTord/src/js/BinarySearchTree.js:20]: Misleading name. Perhaps think about the structure of these functions a bit more.

**Review by KimSvenSand — COMMENTED** [review]
Some changes to be made :octocat:

File: FPBinarySearchTreeTord/src/js/BinarySearchTree.js [hunk 7]
```diff
@@ -0,0 +1,63 @@
...
+    var returnArray = inputArrayInTree(inputArray,0,[]);
+  }
+  else{
+    var returnArray = [];
+  }
+  return returnArray;
+}
+
+export function inputArrayInTree(inputArray,placeInInputArray,returnArray){
+    if(inputArray.length-1 >= placeInInputArray){
```
Comment (emilfolino) [inline FPBinarySearchTreeTord/src/js/BinarySearchTree.js:14]: Muterar returnArray.

File: FPBinarySearchTreeTord/src/js/BinarySearchTree.js [hunk 8]
```diff
@@ -0,0 +1,63 @@
...
+  return returnArray;
+}
+
+export function insert(input,tree){
+  var returnArray = findPlaceForInput(tree,0,input);
+  return returnArray;
+}
+
+export function findNode(input,tree,place,returnValue){
+  if(tree[place] == undefined){
```
Comment (emilfolino) [inline FPBinarySearchTreeTord/src/js/BinarySearchTree.js:40]: muterar returnvalue

File: FPBinarySearchTreeTord/src/js/BinarySearchTree.js (hunk 3 above)
Comment (emilfolino) [reply FPBinarySearchTreeTord/src/js/BinarySearchTree.js:51]: Second that.

**Review by emilfolino — COMMENTED** [review]
Bra jobbat. Skapa kopior istället för mutation.

**Pull request closed without merging**